### ISSUE TITLE
Give 2.0.1 Testing Checklist

### ISSUE THREAD

**Issue by DevinWalker** [WordImpress/Give#2695]
## Issue Overview
I want to be sure every issue reported in 2.0.0  of various database upgrade issues are thoroughly tested before we release:

### Major
- [x] Missing donors due to incomplete database migration
* **Solved:** with a number of upgrade routines that do a health check and then determine upgrade process according to how 2.0.0 progressed.
- [x] Missing form data due to incomplete database migration
* **Solved:** Resolved with health check from above.
- [x] Upgrades with non-standard database prefixes
* **Solved:** Fixed by removing hardcoded `wp_` prefix in upgrade routine
- [x] Stuck upgrade processes where upgrade routine keeps running with no progress
* **Solved:** Added pause/restart functionality so if the rare event something happens there's an easy way to stop the process.

### Minor
- [ ] Incorrect upgrade counts “120481” of “60” completed
* **Not Able to Reproduce:** If someone can reproduce this please email support@givewp.com with your steps to reproduce including database 👍  

- [ ] Invisible upgrade progress bar on the screen - this should be switched from showing the progress of _each_ upgrade to the _overall_ progress.
* **Issue Created** This is a UX issue: https://github.com/WordImpress/Give/issues/2709 we will resolve later. This is mostly happening with small datasets where the updates move really fast.
- [x] False positives for admin-ajax accessibility on certain hosts
* **Solved:** Debugged Flywheel and Pantheon and modified `give_test_ajax_works()` with additional checks and more informative notice pointing to article:  https://givewp.com/documentation/core/frequent-troubleshooting-issues/admin-ajax-blocked/

Please review https://wordpress.org/support/plugin/give for any additional issues I may have left of from above.

## Expected Behavior
Upgrades run smoothly

## Current Behavior
Some customers are experiencing a variety of issues.

## Some Testing Steps

**Scenario 1: Prefixed db**

This is an example of how I expect this issue to be thoroughly tested:

1. Upgrade a site from 1.8.19 to 2.0.0 that has numerous forms, donors, and donations
2. Use a prefixed database 
3. Check if 2.0.0 rollback completed successfully. If it didn't, revert to 1.8.19 and add a number of new donations.
4. Upgrade to 2.0.1 and run upgrade fixes. Confirm that _all_ donations, even the ones from the downgrade to 1.8.19 in the previous step, are displayed properly displaying in both wp-admin and database.

**Comment by ravinderk** [comment]
@DevinWalker 
## 2.0.1 
This contains following fixes:
1. Remove hardcoded table name from WordPress query filters https://github.com/WordImpress/Give/blob/release/2.0.1/includes/class-give-db-meta.php#L304

2. Added pause and restart feature to update. This will allow admin to take over background update process.

3. Few update stalled because payment meta and form meta table are not created on some server. Two automatic updates will fix this issue.
        1. Create meta table ( payment & form )
        2. Check background health and fix if find broken ( missed updates )

**Note:** After upgrading to `2.0.1` ( 10 minutes after plugin update ), if you are seeing the change in update progress then you can force the update to start by pause & restart.

## Testing

### Major
- [x] Missing donors due to incomplete database migration
       **Make sure your backup does not contain table name contains `give_donor` if contains then remove them before starting the upgrade. There is the chance that you will lose newly created donors data. Before deleting tables compare them because if donor upgrade completed then newly created donor will store into new table, so customer table must have less then or equal donors information in donor table**
- [x] Missing form data due to incomplete database migration
- [x] Upgrades with non-standard database prefixes
- [ ] Stuck upgrade processes where AJAX keeps firing
         **On some server ajax is not accessible due to HTTP Authentication,. Most of the admin confuse because notice is not useful to them, We can educate them by improving the message. https://github.com/WordImpress/Give/issues/2697**

### Minor
- [ ] Incorrect upgrade counts “120481” of “60” completed
        **Unable to reproduce.**
- [ ] Invisible upgrade progress bar on the screen - this should be switched from showing the progress of _each_ upgrade to the _overall_ progress.
       **It is the priority. We can fix it later `2.0.1`.**
- [ ] False positives for admin-ajax accessibility on certain hosts
       **Read point 4 under **Major****

I reviewed three issue from wp.org
1. https://wordpress.org/support/topic/db-update-looping-2/ **(need DB, not able to reproduce this)**
2. https://wordpress.org/support/topic/give-cant-update-the-database-because-ajax-is-not-functioning/ **( Read point 4 under **Major** )**
3. https://wordpress.org/support/topic/update-give/ **( If you rolled back your plugin to `1.8.19` then updates will work smoothly in `2.0.1`, tested with dummy DB)**

### Blocker
1. https://github.com/WordImpress/Give/issues/2697 ( @DevinWalker Let me know if you can change ajax notice )

**Note: Please take the backup of the database before and if possible. Update only on staging site if possible then on live site**

**Comment by raftaar1191** [comment]
Working perfectly try updating sites from 1.8.19 to give 2.0.1 the upgrade process was smooth one issues which is already being register https://github.com/WordImpress/Give/issues/2696

**Comment by mathetos** [comment]
I just tested on a customer site, staging environment on WPEngine. I made a video of the whole process, here's what you'll see in it:

1) I updated from 1.8.19 to 2.0 (.org live version) and ran updates and all the donation amounts, donor names, and donor forms were all wiped.
2) I show uploading 2.0.1 manually then running the 2.0.1 database update and the problem still exists
3) For good measure I ran the Recalculate tools just in case
4) I manually updated a donation with a form, donor, and donation amount and saved it and it all saved correctly -- that to me suggests that the data is saved in the correct NEW table, but that old donation information wasn't updated correctly for some reason.

I'm going to use Duplicator to get the database now, and I have credentials for anyone who wants to test this as well.

Here's the video:
http://somup.com/cFV03NVwa4
